pyqtorch/noise.py

```diff
@@ -0,0 +1,356 @@
+from __future__ import annotations
+
+from math import sqrt
+
+import torch
+from torch import Tensor
+
+from pyqtorch.apply import operator_product
+from pyqtorch.matrices import DEFAULT_MATRIX_DTYPE, IMAT, XMAT, YMAT, ZMAT, _dagger
+from pyqtorch.utils import DensityMatrix, density_mat
+
+
+class Noise(torch.nn.Module):
+    def __init__(
+        self, kraus: list[Tensor], target: int, probabilities: tuple[float, ...] | float
+    ) -> None:
+        super().__init__()
+        self.target: int = target
+        self.qubit_support: tuple[int, ...] = (self.target,)
+        self.kraus: list = []
```

Comment:
why is `self.kraus` a list? I suppose they are immutable tensors so they could be stored as buffers?

Reply:
Thank you for your suggestion. I registered the Kraus operators as buffers but I didn’t actually utilize them correctly in the initial implementation.
Thus, I removed `self.kraus` and I updated the `kraus_operator` property to dynamically access the registered buffers instead of relying on a list. This ensures that the tensors are correctly handled by Pytorch: 

```python3
@property
def kraus_operator(self) -> list[Tensor]:
        return [getattr(self, f'kraus_{i}') for i in range(len(self._buffers))]
```